### PR TITLE
Split rest of models

### LINKED ISSUE #72: Split up DB & HTTP requests+response models into packages
> Based on [RFC-0016](https://iver-wharf.github.io/rfcs/published/0016-wharf-api-endpoints-cleanup) (https://github.com/iver-wharf/rfcs/pull/16)

Explained in the RFC:

> Starting with v5, the wharf-api no longer accepts nor returns database models from ANY of its endpoints. Instead, there are now 3 new packages of models inside the wharf-api:
>
> - `pkg/models/database`: /.../
> - `pkg/models/request`: /.../
> - `pkg/models/response`: /.../
>
> https://iver-wharf.github.io/rfcs/published/0016-wharf-api-endpoints-cleanup#requestresponse-models-changes

### PR DESCRIPTION
**Pull request by jilleJr**
- \[x] I've added a new note in the `CHANGELOG.md` file, according to docs:
  https://iver-wharf.github.io/#/development/changelogs/writing-changelogs

## Summary

Lots have been changed here, so bare with me. One by one, I went through the following files and replaced all model usages with the new `pkg/model/database` models, and added models to `pkg/model/request` and `pkg/model/response` on demand:

- artifact.go
- test_result.go
- build.go
- ping.go
- provider.go
- project.go

Ran `swag init` as well as unit and manual tests between each file was converted.

Other changes:

- Removed database_models.go, database_models_json.go, and database_models_test.go
- Added `swaggertype:"object"` to `interface{}` as `swag init` was logging warnings about it
- Added `extensions:"x-nullable"` to `null.*` fields in request & response models

Behaviour should be almost identical to as before, except some more validation that has been added now.

## Motivation

I've intentionally made it so none of the `pkg/model/*` packages depend on each other. The integration between them is done in the `main` package, as can be seen by all the new `dbBranchToResponse`'esque mapping functions.

Another big thing is that the three model packages deal with enums in their own isolated way. While `pkg/model/request` and `pkg/model/response` use strings, the `pkg/model/database` can use ints. And mapping between these are done with `switch` statements, so no actual parsing or "to-string()'ing" is done.

Closes #72.

RFC-0016 states that the `request.{model}Update` models should not contain any IDs to the object itself, but that's deferred to the #70 issue, and this PR does not aim to tackle that one.

### CODE REVIEW

**Review by Alexamakans — APPROVED** [review]

**Comment by jilleJr** [comment]
@Alexamakans There were some changes to deal with the merge conflics from #80 and #82, so I re-requested your review

**Comment by jilleJr** [comment]
Codacy is warning about duplications, but those pop up because I touched the regarding code with my renames and struct usage changes.

I'm not planning on dealing with them, as they are remenant from older code, and will get dealt with in other PRs for the milestone.

Once approved, I will merge the PR despite the failing Codacy check.

**Review by Alexamakans — APPROVED** [review]

File: branch.go [hunk 1]
```diff
@@ -99,80 +108,132 @@ func (m branchModule) createBranchHandler(c *gin.Context) {
...
 		}
 
-		return nil
+		return m.Database.Find(&dbNewBranches).Error
 	})
+
+	if err != nil {
+		log.Error().
+			WithError(err).
+			Message("Failed to replace branch list. Transaction reverted.")
```
Comment (Pikabanga) [inline branch.go:216]: Might as well use SQL nomenclature :)

```suggestion
			Message("Failed to replace branch list. Transaction rolled back.")
```

**Review by Pikabanga — CHANGES_REQUESTED** [review]

File: branch.go [hunk 2]
```diff
@@ -99,80 +108,132 @@ func (m branchModule) createBranchHandler(c *gin.Context) {
...
+			branchNamesSet[reqBranch.Name] = struct{}{}
+			var count int64
+			err := m.Database.
+				Model(&database.Branch{}).
+				Where(&database.Branch{
+					ProjectID: reqBranch.ProjectID,
+					TokenID:   reqBranch.TokenID,
+					Name:      reqBranch.Name,
+				}, database.BranchFields.ProjectID, database.BranchFields.TokenID, database.BranchFields.Name).
+				Limit(1).
```
[the remark this answers is not in the document]
Comment (jilleJr) [reply branch.go:168]: Good point!

File: branch.go (hunk 1 above)
Comment (jilleJr) [reply branch.go]: Very nice!

**Pull request merged**